saleor/warehouse/migrations/0003_warehouse_slug.py

```diff
@@ -8,29 +8,38 @@
 def create_unique_slug_for_warehouses(apps, schema_editor):
     Warehouse = apps.get_model("warehouse", "Warehouse")
 
+    default_slug_value = "warehouse"
     warehouses = (
         Warehouse.objects.filter(slug__isnull=True).order_by(Lower("name")).iterator()
     )
-    previous_char = ""
+    previous_char = None
     slug_values = []
     for warehouse in warehouses:
-        first_char = warehouse.name[0].lower()
-        if first_char != previous_char:
-            previous_char = first_char
-            slug_values = list(
-                Warehouse.objects.filter(slug__istartswith=first_char).values_list(
-                    "slug", flat=True
+        if warehouse.name:
+            first_char = warehouse.name[0].lower()
+            if first_char != previous_char:
+                previous_char = first_char
+                slug_values = list(
+                    Warehouse.objects.filter(slug__istartswith=first_char).values_list(
+                        "slug", flat=True
+                    )
                 )
+        elif previous_char is None:
+            previous_char = ""
+            slug_values = list(
+                Warehouse.objects.filter(
+                    slug__istartswith=default_slug_value
+                ).values_list("slug", flat=True)
             )
 
-        slug = generate_unique_slug(warehouse, slug_values)
+        slug = generate_unique_slug(warehouse, slug_values, default_slug_value)
```

Comment:
`default_slug_value` never changes so it doesn't have to be passed as an argument; it could be a constant at the module level.